tools/odbc/test/tests/connect.cpp

```diff
@@ -1,7 +1,116 @@
-#include "../common.h"
+#include "connect_helpers.h"
+
+#include <iostream>
+#include <odbcinst.h>
 
 using namespace odbc_test;
 
+// Connect to database using SQLDriverConnect without a DSN
+void ConnectWithoutDSN(SQLHANDLE &env, SQLHANDLE &dbc) {
+	std::string conn_str = "";
+	SQLCHAR str[1024];
+	SQLSMALLINT strl;
+
+	SQLRETURN ret = SQLAllocHandle(SQL_HANDLE_ENV, nullptr, &env);
+	REQUIRE(ret == SQL_SUCCESS);
+
+	EXECUTE_AND_CHECK("SQLSetEnvAttr (SQL_ATTR_ODBC_VERSION ODBC3)", SQLSetEnvAttr, env, SQL_ATTR_ODBC_VERSION,
+	                  ConvertToSQLPOINTER(SQL_OV_ODBC3), 0);
+
+	EXECUTE_AND_CHECK("SQLAllocHandle (DBC)", SQLAllocHandle, SQL_HANDLE_DBC, env, &dbc);
+
+	EXECUTE_AND_CHECK("SQLDriverConnect", SQLDriverConnect, dbc, nullptr, ConvertToSQLCHAR(conn_str.c_str()), SQL_NTS,
+	                  str, sizeof(str), &strl, SQL_DRIVER_COMPLETE);
+}
+
+// Connect with incorrect params
+void ConnectWithIncorrectParam(std::string param, std::vector<std::string> expected_msg) {
+	SQLHANDLE env;
+	SQLHANDLE dbc;
+	std::string dsn = "DSN=duckdbmemory;" + param;
+	SQLCHAR str[1024];
+	SQLSMALLINT strl;
+
+	SQLRETURN ret = SQLAllocHandle(SQL_HANDLE_ENV, nullptr, &env);
+	REQUIRE(ret == SQL_SUCCESS);
+
+	EXECUTE_AND_CHECK("SQLSetEnvAttr (SQL_ATTR_ODBC_VERSION ODBC3)", SQLSetEnvAttr, env, SQL_ATTR_ODBC_VERSION,
+	                  ConvertToSQLPOINTER(SQL_OV_ODBC3), 0);
+
+	EXECUTE_AND_CHECK("SQLAllocHandle (DBC)", SQLAllocHandle, SQL_HANDLE_DBC, env, &dbc);
+
+	ret = SQLDriverConnect(dbc, nullptr, ConvertToSQLCHAR(dsn.c_str()), SQL_NTS, str, sizeof(str), &strl,
+	                       SQL_DRIVER_COMPLETE);
+	REQUIRE(ret == SQL_SUCCESS_WITH_INFO);
+
+	std::string state;
+	std::string message;
+	ACCESS_DIAGNOSTIC(state, message, dbc, SQL_HANDLE_DBC);
+	for (const auto &msg : expected_msg) {
+		REQUIRE(duckdb::StringUtil::Contains(message, msg));
+	}
+
+	DISCONNECT_FROM_DATABASE(env, dbc);
+}
+
+// Test sending incorrect parameters to SQLDriverConnect
+static void TestIncorrectParams() {
+	ConnectWithIncorrectParam("UnsignedAttribute=true", {"Invalid keyword", "allow_unsigned_extensions"});
+	ConnectWithIncorrectParam("dtabase=test.duckdb", {"Invalid keyword", "database"});
+	ConnectWithIncorrectParam("this_doesnt_exist=?", {"Invalid keyword"});
+}
+
+// Test setting a database from the connection string
+static void TestSettingDatabase() {
+	SQLHANDLE env;
+	SQLHANDLE dbc;
+
+	auto db_path = "Database=" + GetTesterDirectory() + "test.duckdb";
```

Comment:
In the regular unittests we have a directory specifically for temporary files (see `TestCreatePath`, TestDirectoryPath`) - this directory is deleted after the tests have run, allowing us to dump any files we want in there. Perhaps an idea for the ODBC tests as well?